File: src/main/java/no/nav/syfo/domain/rest/NyttMotebehov.java
package no.nav.syfo.domain.rest;

import lombok.Data;
import lombok.experimental.Accessors;

import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;

@Data
@Accessors(fluent = true, chain = true)
public class NyttMotebehov {

    public String arbeidstakerFnr;

    @NotEmpty
    public String virksomhetsnummer;

    @NotNull
    public MotebehovSvar motebehovSvar;

    public String tildeltEnhet;
}

File: src/main/kotlin/no/nav/syfo/motebehov/NyttMotebehov.kt
package no.nav.syfo.motebehov

import java.io.Serializable
import javax.validation.constraints.NotEmpty

data class NyttMotebehov(
    val arbeidstakerFnr: String? = null,
    val virksomhetsnummer: @NotEmpty String,
    val motebehovSvar: MotebehovSvarLegacyInputDTO,
    val tildeltEnhet: String? = null
) : Serializable
